### PR TITLE
Implement VK_EXT_debug_utils naming

### PR DESCRIPTION
**Pull request by DannyLeWasTaken**
Resolves issue #57

### CODE REVIEW

File: src/core/device.rs [hunk 1]
```diff
@@ -318,6 +320,12 @@ impl Device {
             None
         };
 
+        let debug_utils = if settings.enable_validation {
+            Some( ext::DebugUtils::new( unsafe { instance.loader() }, &instance ))
```
Comment (NotAPenguin0) [inline src/core/device.rs:324]: Run through rustfmt with the provided rustfmt.toml

File: src/core/device.rs [hunk 2]
```diff
@@ -467,6 +476,15 @@ impl Device {
         Ok(self.inner.rt_properties.as_ref().unwrap())
     }
 
+    /// Get the debug_utils object
+    /// # Errors
+    /// - Fails if validation layer is disabled
```
Comment (NotAPenguin0) [inline src/core/device.rs:481]:
```suggestion
    /// - Fails if validation layers are disabled
```

File: src/core/device.rs [hunk 3]
```diff
@@ -467,6 +476,15 @@ impl Device {
         Ok(self.inner.rt_properties.as_ref().unwrap())
     }
 
+    /// Get the debug_utils object
```
Comment (NotAPenguin0) [inline src/core/device.rs:479]:
```suggestion
    /// Get access to the functions of VK_EXT_debug_utils
```

File: src/core/device.rs [hunk 4]
```diff
@@ -467,6 +476,15 @@ impl Device {
...
+    /// # Errors
+    /// - Fails if validation layer is disabled
+    pub fn debug_utils(
+        &self
+    ) -> Result<&ext::DebugUtils> {
+        Ok(self.inner.debug_utils.as_ref().unwrap())
```
Comment (NotAPenguin0) [inline src/core/device.rs:485]: Don't unwrap, instead use `ok_or_else()` to return an error

File: src/core/traits.rs [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+pub trait Nameable: AsRaw {
+	/// Change the name of the given object
+	const OBJECT_TYPE: crate::vk::ObjectType;
+}
+
+impl Device {
+	/// Set the name of any given compatible object for debugging purposes
+	pub fn set_name<T: Nameable>(&self, object: &T, name: &str) -> Result<()> {
+		let object_name = std::ffi::CString::new(name)?;
+		let name_info = crate::vk::DebugUtilsObjectNameInfoEXT::builder()
```
Comment (NotAPenguin0) [inline src/core/traits.rs:22]: Add a `use ash::vk` for consistency with the rest of the crate

File: src/resource/buffer.rs [hunk 6]
```diff
@@ -277,6 +278,16 @@ impl<A: Allocator> Buffer<A> {
     }
 }
 
+unsafe impl crate::core::traits::AsRaw for Buffer {
```
Comment (NotAPenguin0) [inline src/resource/buffer.rs:281]: You can import AsRaw instead of qualifying the type

File: src/resource/buffer.rs [hunk 7]
```diff
@@ -277,6 +278,16 @@ impl<A: Allocator> Buffer<A> {
...
+    unsafe fn as_raw(&self) -> u64 {
+        self.handle().as_raw()
+    }
+}
+
+impl crate::core::traits::Nameable for Buffer {
```
Comment (NotAPenguin0) [inline src/resource/buffer.rs:287]: Same as above

File: src/core/traits.rs [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    unsafe fn as_raw(&self) -> u64;
+}
+
+/// Represents an abstraction for naming objects in Vulkan for debuggers
+pub trait Nameable: AsRaw {
+    /// Change the name of the given object
+    const OBJECT_TYPE: vk::ObjectType;
+}
+
+impl Device {
```
Comment (NotAPenguin0) [inline src/core/traits.rs:20]: Can you move this to be inside the impl block in `device.rs`?

File: src/resource/buffer.rs [hunk 9]
```diff
@@ -35,7 +35,9 @@ use std::ptr::NonNull;
 
 use anyhow::Result;
 use ash::vk;
+use ash::vk::Handle;
```
Comment (NotAPenguin0) [inline src/resource/buffer.rs:38]: unused import

File: src/resource/image.rs [hunk 10]
```diff
@@ -18,7 +18,9 @@ use std::sync::Arc;
 
 use anyhow::Result;
 use ash::vk;
+use ash::vk::Handle;
```
Comment (NotAPenguin0) [inline src/resource/image.rs:21]: unused import

File: src/resource/raytracing/acceleration_structure.rs [hunk 11]
```diff
@@ -2,8 +2,10 @@
 
 use anyhow::Result;
 use ash::vk;
+use ash::vk::Handle;
```
Comment (NotAPenguin0) [inline src/resource/raytracing/acceleration_structure.rs:5]: unused import

File: src/resource/buffer.rs (hunk 9 above)
Comment (DannyLeWasTaken) [reply src/resource/buffer.rs:38]: Used import. Used in .as_raw()

File: src/resource/image.rs (hunk 10 above)
Comment (DannyLeWasTaken) [reply src/resource/image.rs:21]: Used import. Used in .as_raw()

File: src/resource/raytracing/acceleration_structure.rs (hunk 11 above)
Comment (DannyLeWasTaken) [reply src/resource/raytracing/acceleration_structure.rs:5]: Used import. Used in .as_raw()

**Review by NotAPenguin0 — APPROVED** [review]

**Pull request merged**